src/app/pages/pages.module.ts

```diff
@@ -66,11 +63,11 @@ import { SplashModule } from './splash/splash.module'
     UsageService,
     SchemaService,
     NotificationGeneratorService,
-    { provide: NotificationService, useClass: FcmNotificationService },
-    { provide: AnalyticsService, useClass: FirebaseAnalyticsService },
-    FcmNotificationControllerService,
-    RadarUserControllerService,
-    RadarProjectControllerService
+    FcmRestNotificationService,
+    FcmXmppNotificationService,
+    LocalNotificationService,
+    { provide: NotificationService, useClass: NotificationWrapperService },
```

Comment:
Can we not just inject the correct `NotificationService` here instead of having a new service?
You can create a simple factory that provides the correct class based on remote config and use it with an InjectionToken as stated here -- https://angular.io/guide/dependency-injection-in-action#supply-a-custom-provider-with-inject. This will be done on the component level.
If you need to provide the dependency you can use a factory function in the provide clause like follows -
```javascript
{ provide: NOTIFICATION_SERVICE,    useFactory:  notificationServiceFactory(), deps: [RemoteConfigService] }
```

and your function will look something like -
```
export function notificationServiceFactory() {
  return (remoteConfig: RemoteConfigService): NotificationService => {
    /* ... */
  };
};
```

Reply:
Yeah I tried this but there were issues because of the services the notificationService depended on so I just decided to create my own wrapper. But I'll try again.

Reply:
Ok, thanks.

Reply:
I remember now why it wasn't working. It was because of remote config. The remote config returns promises and it's not supported by `useFactory`.

Reply:
Ah, I see, what a shame. Ok, then the above-mentioned updates to the wrapper should be fine. Thanks.